hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/keygen/factory/HoodieSparkKeyGeneratorFactory.java

```diff
@@ -88,6 +88,9 @@ public static KeyGenerator createKeyGenerator(String keyGeneratorClass, TypedPro
         //Need to prevent overwriting the keygen for spark sql merge into because we need to extract
         //the recordkey from the meta cols if it exists. Sql keygen will use pkless keygen if needed.
         && !props.getBoolean(SPARK_SQL_MERGE_INTO_PREPPED_KEY, false);
+    if (autoRecordKeyGen) {
```

Comment:
So `HoodieAvroKeyGeneratorFactory#createAvroKeyGeneratorByType` should be okay?

Reply:
HoodieSparkKeyGeneratorFactory.createKeyGenerator is the one which is called from spark-sql layer, HoodieSparkSqlWriter as well. 

<img width="700" alt="image" src="https://github.com/apache/hudi/assets/513218/4d4862f5-c107-4979-8b89-32ed939e3b9f">

<img width="686" alt="image" src="https://github.com/apache/hudi/assets/513218/5d7f8f73-392d-4011-b662-86cfcf309e2e">

So, we should be good. atleast these are the places where auto key gen will be wrapped.